packages/mui-material/src/styles/unstable_createTheme.js

```diff
@@ -0,0 +1,73 @@
+import { deepmerge } from '@mui/utils';
+import { decomposeColor } from '@mui/system';
+import createThemeWithoutVars from './createTheme';
+import createPalette from './createPalette';
+
+function createTheme(options = {}, ...args) {
+  const { colorSchemes: colorSchemesInput = {}, opacity: opacityInput = {}, ...input } = options;
+
+  // eslint-disable-next-line prefer-const
+  let { palette: lightPalette, ...muiTheme } = createThemeWithoutVars({
+    ...input,
+    ...(colorSchemesInput.light && { palette: colorSchemesInput.light.palette }),
+  });
+  const { palette: darkPalette } = createThemeWithoutVars({
+    palette: { mode: 'dark', ...colorSchemesInput.dark?.palette },
+  });
+
+  colorSchemesInput.light = { palette: lightPalette };
+  colorSchemesInput.dark = { palette: darkPalette };
+
+  const colorSchemes = {};
+
+  Object.keys(colorSchemesInput).forEach((key) => {
+    const palette = createPalette(colorSchemesInput[key].palette);
+
+    Object.keys(palette).forEach((color) => {
+      const colors = palette[color];
+
+      if (colors.main) {
+        palette[color].mainChannel = decomposeColor(colors.main).values.slice(0, 3).join(' ');
```

Comment:
```suggestion
        palette[color].mainChannel = colorChannel(colors.main);
```
Should we create another util function?

Reply:
or `decomposeColor(colors.light).channel`

Reply:
Done, I've added new `colorChannel` method it in the color utils.